html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -86,15 +90,48 @@ public function add()
 		$biditem = $this->Biditems->newEntity();
 		// POST送信時の処理
 		if ($this->request->is('post')) {
-			// $biditemにフォームの送信内容を反映
-			$biditem = $this->Biditems->patchEntity($biditem, $this->request->getData());
+			
+			$file = $this->request->getData('image_path'); //ファイル名受け取り
+
+			$ext = substr($file['name'] , -4);//ファイル名の拡張子だけ切り取る
+			$ext2 = substr($file['name'] , -5);//JPEG用
+			$str = strtoupper($ext);//大文字に変換
+			$str2 = strtoupper($ext2);//JPEG用
+			
+			if($str == '.JPG'||$str == '.GIF'||$str == '.PNG'||$str2== '.JPEG'){
```

Comment:
```suggestion
			$ext = pathinfo($file['name'])['extension'] ?? '';
			$upperExt = strtoupper($ext);
			if (in_array($upperExt, ['JPG', 'GIF', 'PNG', 'JPEG'], true)) {
```
str2とかの処理と変数名がイケてないのでこう書き変えましょう。あと曖昧な比較ではなく厳密な比較を使いましょう